measure/src/measure.rs

```diff
@@ -85,69 +85,31 @@ impl fmt::Display for Measure {
 
 #[cfg(test)]
 mod tests {
-    use {
-        super::*,
-        std::{fmt::Debug, thread::sleep},
-    };
+    use {super::*, std::thread::sleep};
 
     #[test]
     fn test_measure() {
+        let test_duration = Duration::from_millis(100);
         let mut measure = Measure::start("test");
-        sleep(Duration::from_millis(100));
+        sleep(test_duration);
         measure.stop();
-
-        // We have observed failures with margins of 10%, when CI machines are very busy, running
-        // multiple tests in parallel.  As we are not testing the timer functionality itself, it is
-        // probably OK to increase the margins to 20%.
-
-        assert!(measure.as_s() >= 0.08f32 && measure.as_s() <= 0.12f32);
-        assert!(measure.as_ms() >= 80 && measure.as_ms() <= 120);
-        assert!(measure.as_us() >= 80_000 && measure.as_us() <= 120_000);
-        assert!(measure.as_ns() >= 80_000_000 && measure.as_ns() <= 120_000_000);
-        assert!(
-            measure.as_duration() >= Duration::from_millis(90)
-                && measure.as_duration() <= Duration::from_millis(110)
-        );
+        assert!(measure.as_duration() >= test_duration);
```

Comment:
I think this is similar to keeping to old tests, but removing the upper bound.
It is pretty interesting that even 20% was not enough to keep it from failing.

Reply:
> I think this is similar to keeping to old tests, but removing the upper bound.

yeah, it's basically just acknowledging the realities of how `sleep()` works

> It is pretty interesting that even 20% was not enough to keep it from failing.

leaving the accuracy of a test up to system load be like that :wink:

Reply:
> > It is pretty interesting that even 20% was not enough to keep it from failing.
> 
> leaving the accuracy of a test up to system load be like that wink

Thinking a bit more about it, using percentages is not necessarily correct.
There are a number of factors that are considered by the scheduler when it chooses the next thread, and the length of the sleep does not actually matter.
At least on Windows, only the length of the wait since the required activation matters.
I am not familiar enough with the current Linux scheduler.
But, I would think, that Linux scheduler also does not care how long was the thread was sleeping.

So, to compensate, it would probably be more accurate to talk about absolute values.
Such as "results are good if accurate within 100ms of the expected", rather than "within 20% of expected".
With 100ms sleep times, 20% is 20ms.

The delay will be affected by how many other threads are waiting to be scheduled.  CFS, the current Linux scheduler, seems to be [reducing the slice time to 1ms](https://notes.eddyerburgh.me/operating-systems/linux/process-scheduling#fair-scheduling).
So if the test shows an error over 20ms, there are 20 threads that need to be scheduled on the same core.  CI runs on a 48 core system.

So there are probably more than 960 threads that are competing for the CPU.
Seems like a lot :)

Reply:
or we could just... not test code that isn't ours.  its not like there's anything that a `Measure` does that we could use to compensate for any discrepancy of tested bounds.  testing the upper bound was simply the wrong thing to do, period.  the new test just ensures that the timer has actually advanced by at least the sleep period, which is the only guarantee that sleep makes (+/- signals or thread cancellation)

Reply:
Absolutely, it is better if the scope can just be decreased :)